index.bs

```diff
@@ -1533,7 +1533,155 @@ like {{WeakRef}} or {{FinalizationRegistry}},
+
+Dictionaries are "passed by value" to methods (i.e., they are copied).
+Browsers engines strip unknown members when converting from JavaScript objects to a WebIDL representation.
+This means that changing the value after it is passed into an API has no effect.
+
+Again, taking the [`ShareData`](https://www.w3.org/TR/web-share/#dom-sharedata) dictionary as an example:
+
+```JS
+const data = {
+    "text": "Text being shared",
+    // Ignored by a browser that does not include a "whatever" parameter.
+    "whatever": 123,
+};
+let p = navigator.share(data);
+
+// Changing this after calling .share() has no effect
+data.text = "New text";
+```
+
+<h4 class="no-num no-toc">Choose an Interface for Functionality, State, and Identity</h4>
+
+Interfaces are roughly equivalent to classes in JavaScript.
+Use an interface when a specification needs to bundle state--
+both visible properties and internal "slots"--
+with operations on that state (i.e., methods).
+
+Unlike dictionaries, interfaces:
+
+ * can have instances with state,
+ * can have need read-only properties,
+ * can exhibit side-effects on assignment, and
+ * provide the ability to check object's identity
+    (i.e., one can check if  it is an `instanceof` a particular class on the global scope),
+
+Defining an interface also exposes it on the global scope, allowing for the specification of static methods.
+For example, the `canParse()` static method of the URL interface. 
+
+```JS
+if (URL.canParse(someURL)) {
+    // Do stuff...
+}
+```
+
+Give stateful interfaces a constructor, if possible.
+Do not add a constructor if a class has no state.
+Doing so is considered *bad practice*, as it is effectively creating a "fake class":
+that is, instances of the interface do nothing that a static method couldn't do.
+[`DOMParser`](https://html.spec.whatwg.org/#domparser) or [`DOMImplementation`](https://dom.spec.whatwg.org/#domimplementation) are examples of fake classes.
+
+<h4 class="no-num no-toc">Provide a serializer to make interface data more accessible</h4>
+
+Add serializers to transform instances of an interface into a form that is expected to be used by many applications.
+
+A `.toJSON()` method allows an instance of an interface to produce a useful JSON serialization.
+A `.toBlob()` method can be used to extract a binary representation of an interface.
+
+This makes object natural to use with APIs.
+For example, [`GeolocationPosition`](https://www.w3.org/TR/geolocation/#dom-geolocationposition)
+provides a `toJSON()` method:
+
+```JS
+const position = await new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition(resolve, reject);
+});
+
+const message = JSON.stringify({
+    user: userId,
+    time: Date.now(),
+    position, // .stringify() calls .toJSON() automatically
+});
+```
+
+<h4 class="no-num no-toc">Choose a namespace to Avoid “Fake Classes” for Behavior-Only Utilities</h4>
```

Comment:
```suggestion
<h4 id="namespace-to-avoid-fake-classes" class="no-num no-toc">Choose a namespace to Avoid “Fake Classes” for Behavior-Only Utilities</h4>
```